machine/accton/accton_as_28t_ops/u-boot/platform-as_28t_ops.patch

```diff
@@ -0,0 +1,2573 @@
+ hurricane2_noshmoo           arm         armv7       bcm95615x           broadcom       hurricane2        hurricane2:CUSTOM_LINKER_SCRIPT
+ hurricane2_rom_emul          arm         armv7       bcm95333x           broadcom       hurricane2        hurricane2_rom_emul:L2C_AS_RAM,NO_CODE_RELOC
++AS_28T_OPS                   arm         armv7       as_28t_ops           accton        hurricane2        AS_28T_OPS:ONIE_PLATFORM_REV=0
+ katana2                       arm         armv7       bcm95645x           broadcom       katana2           katana2:RUN_DDR_SHMOO2,IPROC_DDR_ECC 
+ katana2_nand                  arm         armv7       bcm95645x           broadcom       katana2           katana2:RUN_DDR_SHMOO2,IPROC_DDR_ECC,NAND_IPROC_BOOT
+ origen			     arm	 armv7	     origen		 samsung	exynos
+diff --git a/broadcom_boards.mk b/broadcom_boards.mk
+index 60b2ecf..ba75cdf 100644
+--- a/broadcom_boards.mk
++++ b/broadcom_boards.mk
+@@ -151,6 +151,10 @@ hurricane2_rom_emul_config: $(obj)hurricane2_rom_emul_config
+ $(obj)hurricane2_rom_emul_config: $(TOPDIR)/include/configs/hurricane2_rom_emul.h 
+ 	@$(MKCONFIG) -A $(@F:_config=); touch $@
+ 
++AS_28T_OPS_config: $(obj)AS_28T_OPS_config
++$(obj)AS_28T_OPS_config: $(TOPDIR)/include/configs/AS_28T_OPS.h
++	@$(MKCONFIG) -A $(@F:_config=); touch $@
++
+ katana2_config: $(obj)katana2_config 
+ $(obj)katana2_config: $(TOPDIR)/include/configs/katana2.h  
+ 	@$(MKCONFIG) -A $(@F:_config=); touch $@
+diff --git a/common/cmd_sys_eeprom.c b/common/cmd_sys_eeprom.c
+index cf2ed1c..6a52b5e 100644
+--- a/common/cmd_sys_eeprom.c
++++ b/common/cmd_sys_eeprom.c
+@@ -1206,11 +1206,29 @@ int mac_read_from_eeprom(void)
+ 				macbase[0], macbase[1], macbase[2],
+ 				macbase[3], macbase[4], macbase[5]);
+ 			sprintf(enetvar, i ? "eth%daddr" : "ethaddr", i);
++#if 1
++{
++			/* aaron_lien, 2013/08/05,
++			 * Initialize $ethaddr if it is blank or
++			 * if the current value is different from non-volatile board information
++			 * then change it.
++			 */
++			char *env;
++			env = getenv(enetvar);
++			if (!env) {
++				setenv(enetvar, ethaddr);
++			} else {
++				if(strcmp(env, ethaddr))
++					setenv(enetvar, ethaddr);
++			}
++}
++#else
+ 			/* Only initialize environment variables that are blank
+ 			 * (i.e. have not yet been set)
+ 			 */
+ 			if (!getenv(enetvar))
+ 				setenv(enetvar, ethaddr);
++#endif
+ 
+ 			macbase[5]++;
+ 			if (macbase[5] == 0) {
+diff --git a/common/env_common.c b/common/env_common.c
+index d6162a4..9387b67 100644
+--- a/common/env_common.c
++++ b/common/env_common.c
+@@ -31,6 +31,7 @@
+ #include <search.h>
+ #include <errno.h>
+ #include <malloc.h>
++#include <spi_flash.h>
+ 
+ DECLARE_GLOBAL_DATA_PTR;
+ 
+@@ -328,10 +329,35 @@ int env_complete(char *var, int maxv, char *cmdv[], int bufsz, char *buf)
+ */
+ void env_write_default_if_crc_bad(void)
+ {
+-	env_t *env_ptr = (env_t *)CONFIG_ENV_ADDR;
++	int ret;
```

Comment:
I think we can move this SPI env code to the common patch also.

Just protect the code with:
```
#ifdef CONFIG_ENV_IS_IN_SPI_FLASH
/* new code here */
#else
/* old code here */
#endif